### PR TITLE
add-bastion-subnet-whitelist

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/22771

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- Add subnet whitelist for bastion SSH port 22.
```
Comment (fiunchinho) [inline CHANGELOG.md:11]:
```suggestion

- Add subnet whitelist for bastion SSH port 22.
```

File: helm/cluster-gcp/templates/_gcp_cluster.tpl [hunk 2]
```diff
@@ -2,6 +2,8 @@
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
 kind: GCPCluster
 metadata:
+  annotations:
+    bastion.gcp.giantswarm.io/whitelist: {{ .Values.bastion.whitelistSubnets }}
```
Comment (fiunchinho) [inline helm/cluster-gcp/templates/_gcp_cluster.tpl:6]: What would be consuming this annotation?

**Review by fiunchinho — CHANGES_REQUESTED** [review]
I would prefer to use `allowlist` rather than `whitelist`, as it's clearer what it purpose is.

File: helm/cluster-gcp/templates/_gcp_cluster.tpl (hunk 2 above)
Comment (calvix) [reply helm/cluster-gcp/templates/_gcp_cluster.tpl:6]: `capg-firewall-rule-operator`

**Comment by calvix** [comment]
`whitelist` is a very commonly used term for such configuration, even the ingress controller uses the term, we use it in vintage, I don't want to use something different than  the k8s world is use to

example for IC
```
    nginx.ingress.kubernetes.io/whitelist-source-range: 49.36.X.X/32
```

**Comment by calvix** [comment]
```
A whitelist (or, less commonly, a passlist or allowlist) is a mechanism which explicitly allows some identified entities to access a particular privilege, service, mobility, or recognition i.e. it is a list of things allowed when everything is denied by default.
```

**Comment by fiunchinho** [comment]
One of the benefits of using `allowlist` over `whitelist` is that you don't need to share the definition because the meaning is already in the name. Since we are not using the exact same annotation than nginx uses but creating a new one, I think it's not that important what nginx uses.

**Comment by fiunchinho** [comment]
Also, [k8s seems to be moving on the same direction](https://github.com/kubernetes/kubernetes/issues/90277) so I'd say it would be going against upstream direction.

**Comment by mnitchev** [comment]
Implements giantswarm/roadmap#1233

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**